source/visionEnhancementProviders/screenCurtain.py

```diff
@@ -300,6 +300,46 @@ def confirmInitWithUser(self) -> bool:
 			return res == wx.YES
 
 
+def isScreenFullyBlack() -> bool:
+	"""
+	Uses wx to check that the screen is currently fully black by taking a screen capture and checking:
+	- there is only one colour used in the image
+	- the first pixel of the screen capture is black
+
+	Having to iterate over a large bitmap image in python is slow (~4s depending on machine, number of pixels).

```

Comment:
Can you make this a little more specific about the alternative tested. If the reader isn't aware of the `screenBitmap` module then this will be confusing since the `wx.Bitmap` is being used.

Perhaps something like:
```
wx Python doesn't support getting the color values back from the histogram. 
As an alternative, NVDA's 'screenBitmap' module was tested but found to be too slow. 
Fetching the bitmap with the 'screenBitmap' module was fast, but iterating over
it with `<insert code sample here>` to check values was slow (~4 seconds)
```

Reply:
This is a general problem though, iterating over all the pixels (for modern display sizes) in python is going to be slow.

>wx Python doesn't support getting the color values back from the histogram.

Reply:
>wx Python doesn't support getting the color values back from the histogram. 

I don't think this is a related problem - we could also iterate over a `wx.Bitmap`. The histogram was used to improve performance.

Reply:
It's just to explain the reasons why the approach might be considered. Adding this information in comments prevents future developers from investigating the same dead ends as you.

Reply:
Explaining how this solution was arrived at, and considered the best way forward at the time.

Reply:
I've addresses this now.